sample-app/src/main/java/com/joinforage/android/example/ui/pos/POSComposeApp.kt

```diff
@@ -209,15 +216,115 @@ fun POSComposeApp(
                     onDoneButtonClicked = { navController.popBackStack(POSScreen.ActionSelectionScreen.name, inclusive = false) }
                 )
             }
-            composable(route = POSScreen.PaymentTypeSelectionScreen.name) {
+            composable(route = POSScreen.PAYTransactionTypeSelectionScreen.name) {
                 PaymentTypeSelectionScreen(
-                    onSnapPurchaseClicked = { /*TODO*/ },
+                    onSnapPurchaseClicked = { navController.navigate(POSScreen.PAYSnapPurchaseScreen.name) },
                     onCashPurchaseClicked = { /*TODO*/ },
                     onCashWithdrawalClicked = { /*TODO*/ },
                     onCashPurchaseCashbackClicked = { /*TODO*/ },
                     onCancelButtonClicked = { navController.popBackStack(POSScreen.ActionSelectionScreen.name, inclusive = false) }
                 )
             }
+            composable(route = POSScreen.PAYSnapPurchaseScreen.name) {
+                EBTSnapPurchaseScreen(
+                    onConfirmButtonClicked = { snapAmount ->
+                        val payment = PosPaymentRequest(
+                            amount = snapAmount,
+                            description = "Testing POS certification app payments (SNAP Purchase)",
+                            fundingType = FundingType.EBTSnap.value,
+                            paymentMethodRef = "",
+                            posTerminal = PosTerminal(providerTerminalId = k9SDK.terminalId),
+                            metadata = mapOf()
+                        )
```

Comment:
Food for thought: maybe an opportunity to abstract away some of these default values to keep the `POSComposeApp` a bit lighter. Certainly not a requirement

```kotlin
// example usage
EBTSnapPurchaseScreen(
    onConfirmButtonClicked = { 
        snapAmount -> PosPaymentRequest.forSnapPayment(
            snapAmount,
            k9SDK.terminalId,
        ) 
    }
```

```kotlin
// example implementation
@JsonClass(generateAdapter = true)
data class PosPaymentRequest(
    val amount: Double,
    @Json(name = "funding_type") val fundingType: String,
    @Json(name = "payment_method") val paymentMethodRef: String,
    val description: String,
    @Json(name = "pos_terminal") val posTerminal: PosTerminal,
    val metadata: Map<String, String>
) {
    companion object {
        fun forSnapPayment(snapAmount: Int, terminalId: String) = PosPaymentRequest(
            amount = snapAmount,
            description = "Testing POS certification app payments (SNAP Purchase)",
            fundingType = FundingType.EBTSnap.value,
            paymentMethodRef = "",
            posTerminal = PosTerminal(terminalId),
            metadata = mapOf()
        )
    }
}
```

Reply:
Yeah I like this – was thinking there was probably a cleaner way and I think this makes a lot of sense